plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1955,10 +1955,10 @@ protected void setVmBootDetails(final VM vm, final Connection conn, String bootT
         boolean isSecure = bootType.equals(ApiConstants.BootType.UEFI.toString()) &&
                 ApiConstants.BootMode.SECURE.toString().equals(bootMode);
         final Map<String, String> bootParams = vm.getHVMBootParams(conn);
-        bootParams.replace("firmware", bootType);
+        bootParams.replace("firmware", bootType.toLowerCase());
         vm.setHVMBootParams(conn, bootParams);
         final Map<String, String> platform = vm.getPlatform(conn);
-        platform.put("secureboot", isSecure ? "true" : "false");
+        platform.put("secureboot", Boolean.toString(isSecure));
```

Comment:
How about making isSecure `Boolean` instead of `boolean` and than just call isSecure.teString()

Reply:
Not sure how that helps or improves performance. Unless the point is to make code look good in a certain way.
And that has been the whole point of reviews lately IMO. 
Anyway, changed it before it becomes an argument and RM has to change it
![Screenshot from 2021-08-21 00-07-52](https://user-images.githubusercontent.com/153340/130279777-ae307e2e-5f2a-4d62-ba30-574f2cb68b44.png)